lib/cli-engine.js

```diff
@@ -391,12 +393,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options || {});
+    options.cwd = options.cwd || process.cwd();
```

Comment:
Be careful - this could be modifying the `options` object that is passed in. Best to use `assign()` to create a copy before modifying (like line 403).